script/watch.js

```diff
@@ -0,0 +1,103 @@
+const chokidar = require('chokidar');
+const CLIEngine = require('eslint').CLIEngine;
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+
+const esLinter = new CLIEngine();
+const sasslinter = require('sass-lint');
+
+const chalk = require('chalk');
+const path = require('path');
+const srcWatcher = chokidar.watch(['./src/**/*.jsx', './src/**/*.js', './src/**/*.scss']);
+
+let running = false;
+let restart = false;
+
+function formatter(eslintResults) {
+  return eslintResults.reduce((acc, current) => {
+    acc.push(chalk.cyan(`${path.basename(current.filePath)}\n`));
+    current.messages.forEach((message) => {
+      acc.push(chalk.cyan(`  line ${message.line} `));
+      acc.push(chalk.red(`${message.message}\n`));
+    });
+    return acc;
+  }, [])
+};
+
+async function getChangedFiles() {
+  const gitDiffIndex = await exec('git diff-index --name-only HEAD');
+  return gitDiffIndex.stdout.split('\n');
+}
+
+async function runLints() {
+  console.log('runLints');
+  if (running) {
+    restart = true;
+    return;
+  } else {
+    running = true;
+    restart = false;
```

Comment:
Will probably ditch this for debouncing. Haven't tested with changing branches yet- I don't want this to blow up with a bunch of recursive calls to git after a branch change.